docs/package_templates/autotools_package/all/conanfile.py

```diff
@@ -75,64 +85,99 @@ def requirements(self):
 
     def validate(self):
         # validate the minimum cpp standard supported. Only for C++ projects
-        if self.info.settings.compiler.cppstd:
+        if self.info.settings.compiler.get_safe("cppstd"):
             check_min_cppstd(self, 11)
         if self.info.settings.os not in ["Linux", "FreeBSD", "MacOS"]:
             raise ConanInvalidConfiguration(f"{self.ref} is not supported on {self.info.settings.os}.")
 
     # if another tool than the compiler or autotools is required to build the project (pkgconf, bison, flex etc)
     def build_requirements(self):
+        # only if we have to call autoreconf
         self.tool_requires("libtool/x.y.z")
-        self.tool_requires("pkgconf/x.y.z")
+        # only if upstream configure.ac relies on PKG_CHECK_MODULES macro
+        if not self.conf.get("tools.gnu:pkg_config", default=False, check_type=str):
+            self.tool_requires("pkgconf/x.y.z")
```

Comment:
I've taken a look at conan source code, and conan v2 build helpers don't seem to do any special management of `tools.gnu:pkg_config`, so they don't guarantee that this path is discoverable.